ksqldb-engine/src/test/java/io/confluent/ksql/services/DefaultConnectClientFactoryTest.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.services;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.confluent.ksql.test.util.KsqlTestFolder;
+import io.confluent.ksql.util.KsqlConfig;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.Charset;
+import java.util.Base64;
+import java.util.Optional;
+import org.apache.kafka.common.config.ConfigException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DefaultConnectClientFactoryTest {
```

Comment:
If it is not too much trouble, I'd probably test that credentials are reloaded on file change.

Reply:
I didn't add one initially since I thought it'd have to be an integration test, rather than a unit test, which would be a fair bit of work but now that I'm thinking about it again I think we can get away with a unit test. If so, I'll add one. If not, I'd rather not block on this. (We already have plans for integration tests for this functionality elsewhere where the configs will be exercised.)

Reply:
Added unit tests for reload in the latest commit.